### PR TITLE
Adding support for SSD1306 OLED display with 7-pin I2C/SPI

### PR DESCRIPTION
**Pull request by devdotnetorg**
The existing code only supports SSD1306 display with 4-pin.
But there is a version with 7-pin I2C/SPI. For this version it is necessary to initialize the pin - RES.

## Description
Adding support for SSD1306 OLED display with 7-pin I2C/SPI for only I2C.

Description will be available on June 13-17 at the link - [post (rus)](https://habr.com/ru/company/timeweb/blog/669700/).

## Motivation and Context

- Fixes

## How Has This Been Tested?

- VS 2019 Community Edition on Windows 10
- ESP-WROOM-32

## Screenshots
[IMAGE_ALT](https://devdotnet.org/wp-content/media/new-nf-lcd-wifi-title.jpg)

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [x] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [x] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [x] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/nanoframework/nanoFramework.IoT.Device?pullRequest=443) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/nanoframework/nanoFramework.IoT.Device?pullRequest=443) before we can accept your contribution.<br/><br/>:x: devdotnetorg [sign now](https://cla.dotnetfoundation.org/nanoframework/nanoFramework.IoT.Device?pullRequest=443)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/nanoframework/nanoFramework.IoT.Device?pullRequest=443) it.</sub>

File: devices/Ssd13xx/Ssd1306.cs [hunk 1]
```diff
@@ -28,17 +33,42 @@ public Ssd1306(I2cDevice i2cDevice) : base(i2cDevice)
...
 
+        /// <summary>
+        /// Initializes new instance of Ssd1306 device that will communicate using I2C bus.
+        /// A single-chip CMOS OLED/PLED driver with controller for organic/polymer
+        /// light emitting diode dot-matrix graphic display system.
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device used for communication.</param>
+        /// <param name="rstPinNumber">Reset pin (some displays might be wired to share the microcontroller's
+        /// reset pin).</param>
+        public Ssd1306(I2cDevice i2cDevice, int rstPinNumber) : base(i2cDevice, rstPinNumber)
```
Comment (Ellerbach) [inline devices/Ssd13xx/Ssd1306.cs:44]: If you pass a pin number, you need to allow the possibility to pass the GPIO controller and also add the shouldDispose boolean to dispose it or not at the end. You'll find the pattern in other bindings.

File: devices/Ssd13xx/Ssd13xx.cs [hunk 2]
```diff
@@ -527,5 +533,23 @@ public enum DisplayResolution
...
         };
+
+        /// <summary>
+        /// Reset display controller.
+        /// </summary>
+        /// <param name="rstPinNumber">Reset pin (some displays might be wired to share the microcontroller's
+        /// reset pin).</param>
+        private void Reset(int rstPinNumber)
+        {
+            var s_GpioController = new GpioController();
```
Comment (Ellerbach) [inline devices/Ssd13xx/Ssd13xx.cs:544]: As pointed, this is not the correct pattern. Please use the proper pattern.
Also, you need to check that rstPinNumber is valid.

**Review by Ellerbach — CHANGES_REQUESTED** [review]
Please adjust the GpioController to the proper pattern. And it will be good!

**Comment by josesimoes** [comment]
@devdotnetorg any chance you can resume work on this? It can get a bit behind the develop branch and it's more work... 😉

**Comment by devdotnetorg** [comment]
@josesimoes Yes, I will finish the work, but only at the end of this week.

**Comment by Ellerbach** [comment]
@devdotnetorg as we're now a bit more than the end of the week you mentioned, any chance to move it forward? Thanks!

**Comment by devdotnetorg** [comment]
@Ellerbach I will likely close the change-ssd13xx branch and create a new one to avoid merge issues

**Comment by Ellerbach** [comment]
@devdotnetorg are you still working on this? Any help needed?

**Comment by devdotnetorg** [comment]
@Ellerbach I created a theme [Adding support for SSD1306 OLED display with 7-pin I2C/SPI V2.](https://github.com/nanoframework/nanoFramework.IoT.Device/pull/550)

**Pull request closed without merging**